### ISSUE TITLE
kubectl create secret with docker-registry creates a secret that can't be used

### ISSUE THREAD

**Issue by nicolls1** [kubernetes/kubectl#336]
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): Bug

**Kubernetes version** (use `kubectl version`):
Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.0", GitCommit:"925c127ec6b946659ad0fd596fa959be43f0cc05", GitTreeState:"clean", BuildDate:"2017-12-15T21:07:38Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"", Minor:"", GitVersion:"v1.9.0", GitCommit:"925c127ec6b946659ad0fd596fa959be43f0cc05", GitTreeState:"clean", BuildDate:"2018-01-26T19:04:38Z", GoVersion:"go1.9.1", Compiler:"gc", Platform:"linux/amd64"}

**Environment**:
- **Cloud provider or hardware configuration**:Minikube
- **OS** (e.g. from /etc/os-release):Ubuntu 16.04
- **Kernel** (e.g. `uname -a`):Linux barley 4.4.0-62-generic #83-Ubuntu SMP Wed Jan 18 14:10:15 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
- **Install tools**:
- **Others**:

**What happened**: I followed the guild lines [here](https://kubernetes.io/docs/concepts/containers/images/#configuring-nodes-to-authenticate-to-a-private-repository) for accessing a private docker image repo. The command `kubectl create secret docker-registry myregistrykey ...` created a secret that looked like the following when printing in yaml format:
```
apiVersion: v1
data:
  .dockercfg: *******
kind: Secret
metadata:
  creationTimestamp: 2018-03-03T19:48:59Z
  name: regcred
  namespace: default
  resourceVersion: "6564"
  selfLink: /api/v1/namespaces/default/secrets/regcred2
  uid: ea913199-1f1b-11e8-8166-080027d635ca
type: kubernetes.io/dockercfg
```

When i added regcred to my deployment imagePullSecret I was still getting pull errors. After looking around and trying a bunch of different things I finally got the following to work.
```
apiVersion: v1
kind: Secret
metadata:
  name: regcred
data:
  .dockerconfigjson: *****
type: kubernetes.io/dockerconfigjson
```

I used the exact same base64 string in both and the difference seems to be just the `type: kubernetes.io/dockerconfigjson` field where that is the one that works while the command uses the `type: kubernetes.io/dockercfg` type.

**What you expected to happen**: I expected the secret to be generated in the correct format

**How to reproduce it** (as minimally and precisely as possible): Run `kubectl create secret docker-registry myregistrykey ...`and see that generated secret uses `type: kubernetes.io/dockercfg`

**Comment by mengqiy** [comment]
We used to support both.
It seems kubelet drop the support for `.dockercfg`.
@Random-Liu Can you confirm?

**Comment by liggitt** [comment]
this was a bug in 1.9.0, fixed in 1.9.1

https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.9.md#changelog-since-v190

> Fixes issue creating docker secrets with kubectl 1.9 for accessing docker private registries. (#57463, @dims)
